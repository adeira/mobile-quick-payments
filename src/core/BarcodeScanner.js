// @flow

import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { BarCodeScanner, Permissions } from 'expo';
import { Actions } from 'react-native-router-flux';

type State = {|
  hasCameraPermission: boolean | null,
|};

export default class BarcodeScanner extends React.Component<{||}, State> {
  state = {
    hasCameraPermission: null,
  };

  componentWillMount = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA);
    this.setState({ hasCameraPermission: status === 'granted' });
  };

  handleBarCodeRead = ({ data }: { data: string }) => {
    Actions.payment();
  };

  render = () => {
    const { hasCameraPermission } = this.state;

    if (hasCameraPermission === null) {
      return <Text>Requesting for camera permission</Text>;
    } else if (hasCameraPermission === false) {
      return <Text>No access to camera</Text>;
    } else {
      return (
        <View style={{ flex: 1 }}>
          <BarCodeScanner
            onBarCodeRead={this.handleBarCodeRead}
            barCodeTypes={[BarCodeScanner.Constants.BarCodeType.qr]}
            style={StyleSheet.absoluteFill}
          />
        </View>
      );
    }
  };
}
