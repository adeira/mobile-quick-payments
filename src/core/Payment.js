// @flow

import * as React from 'react';
import { Title, SafeAreaView } from 'mobile-quick-payments-shared';

export default class Payment extends React.Component<{||}> {
  render = () => (
    <SafeAreaView>
      <Title>How much?</Title>
    </SafeAreaView>
  );
}
