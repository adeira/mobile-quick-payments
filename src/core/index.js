// @flow

module.exports = {
  Homepage: require('./Homepage').default,
};
