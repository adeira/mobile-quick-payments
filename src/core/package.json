{
  "name": "mobile-quick-payments-core",
  "version": "0.0.0",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "mobile-quick-payments-shared": "^0",
    "react-native-router-flux": "^4.0.0-beta.28"
  }
}
