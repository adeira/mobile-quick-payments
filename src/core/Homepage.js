// @flow

import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Title, SafeAreaView } from 'mobile-quick-payments-shared';

import BarcodeScanner from './BarcodeScanner';

export default class Homepage extends React.Component<{||}> {
  render = () => (
    <SafeAreaView style={style.barcodeScanner}>
      <Title>Scan QR code to pay</Title>
      <BarcodeScanner />
    </SafeAreaView>
  );
}

const style = StyleSheet.create({
  barcodeScanner: {
    flex: 1,
  },
});
