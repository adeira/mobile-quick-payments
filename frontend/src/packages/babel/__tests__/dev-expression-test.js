// @flow

import * as babel from '@babel/core';

function transform(input) {
  return babel.transform(input, {
    plugins: [require('../dev-expression')],
  }).code;
}

let oldEnv;

// FIXME: this is broken now but otherwise everything work
xdescribe('dev-expression', () => {
  beforeEach(() => {
    oldEnv = process.env.NODE_ENV;
    process.env.NODE_ENV = '';
  });

  afterEach(() => {
    process.env.NODE_ENV = oldEnv;
  });

  it('should replace __DEV__ in if', () => {
    expect(transform("if (__DEV__) { console.log('foo') }")).toMatchSnapshot();
  });

  it('should replace __DEV__ in ternary operator', () => {
    expect(transform('const dev = __DEV__ ? true : false')).toMatchSnapshot();
  });
});
