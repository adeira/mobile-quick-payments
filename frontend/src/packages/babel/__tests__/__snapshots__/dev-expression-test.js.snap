// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dev-expression should replace __DEV__ in if 1`] = `
"if (process.env.NODE_ENV !== \\"production\\") {
  console.log('foo');
}"
`;

exports[`dev-expression should replace __DEV__ in ternary operator 1`] = `"const dev = process.env.NODE_ENV !== \\"production\\" ? true : false;"`;
